Code.gs

```diff
@@ -25,6 +25,8 @@ var sourceCalendars = [                // The ics/ical urls that you want to get
                                        // For instance: ["https://p24-calendars.icloud.com/holidays/us_en.ics", "US Holidays"]
                                        // Or with colors following mapping https://developers.google.com/apps-script/reference/calendar/event-color,
                                        // for instance: ["https://p24-calendars.icloud.com/holidays/us_en.ics", "US Holidays", "11"]
+                                       // Login and password can be added
+                                       // for instance: ["https://login:password@p24-calendars.icloud.com/holidays/us_en.ics", "US Holidays"]
```

Comment:
The question imo is, what's the benefit of importing a 2365 loc library compared to checking/extracting username:password with a 'simple' [regex](https://codepal.ai/regex-generator/query/h87ZYdbN/regex-url-with-username-password-domain-port-path)?

Reply:
Or, better than using a regex where we probably shouldn't, We could add two more optional settings:

```
var authUsername = "";
var authPassword = "";
```

Or even a single `var authString = "";` that is just the literal auth header value (eg `Bearer ...`)

Reply:
The credentials need to be url specific though, not sure if we want to add another parameter to the arrays.

Reply:
True. This might look better once I prototype #311.

@tatankat - can you make the changes discussed here? Since this seems to be an agreed-upon standard, let's keep the URI format. However, let's prefer using a regex (@jonas0b1011001 linked one example above - but you might not need one that complicated. I also like to encourage the usage of [named capturing groups](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Named_capturing_group) when possible) rather than including the URI.js.gs additional library.

Reply:
There you go. I thought using a library would make it more robust.
I also stumbled on a limitation of this notation needing percent-encoding for certain characters. Additional parameters might still be a good idea.